app/generate-index/gen-site-data.ts

```diff
@@ -56,19 +58,28 @@ export async function getHtmlPages(urlList: string[]) {
 // TODO: make this script more fault tolerant w retry behavior.
 // right now fails when there are momentary internet drops
 export async function getPageData(url: string) {
+  axiosRetry(axios, { retryDelay: axiosRetry.exponentialDelay });
   const { data: htmlPage } = await axios.get(url);
   return { url, htmlPage };
 }
 
-export function snootyHtmlToText(html: string) {
+function getPageBody(html: string) {
   const dom = new JSDOM(html);
   const $ = require("jquery")(dom.window);
 
-  $("html").find("header").remove();
-  $("html").find("footer").remove();
-  $("html").find("nav").remove();
+  const content = $("main").html();
```

Comment:
looking at the MD output, all the headers are postfixed with the string `[![](<path-to-site>/assets/link.svg)](#introduction \"Permalink to this heading\")` 

could you remove this in the parsing?

Reply:
There are plenty of issues with the parsing that aren't trivial to solve in this PR, working on them in another branch.